TestConfigTestSuite is unstable

## Flaky Test
TestConfigTestSuite

### Which jobs are failing
```bash
2023-11-24T09:13:41.3004455Z --- FAIL: TestConfigTestSuite (59.50s)
2023-11-24T09:13:41.3004734Z     --- FAIL: TestConfigTestSuite/TestConfig (7.51s)
2023-11-24T09:13:41.3005090Z         testutil.go:265: start to run test in pd mode
2023-11-24T09:13:41.3005419Z         testutil.go:269: finish to run test in pd mode
2023-11-24T09:13:41.3005674Z         testutil.go:274: start to run test in api mode
2023-11-24T09:13:41.3005960Z         config_test.go:109: 
2023-11-24T09:13:41.3006490Z             	Error Trace:	/home/runner/work/pd/pd/tests/pdctl/config/config_test.go:109
2023-11-24T09:13:41.3007316Z             	            				/home/runner/work/pd/pd/tests/testutil.go:279
2023-11-24T09:13:41.3007985Z             	            				/home/runner/work/pd/pd/tests/testutil.go:260
2023-11-24T09:13:41.3009023Z             	            				/home/runner/work/pd/pd/tests/pdctl/config/config_test.go:64
2023-11-24T09:13:41.3009346Z             	Error:      	Should be false
2023-11-24T09:13:41.3009653Z             	Test:       	TestConfigTestSuite/TestConfig
```

### CI link
https://github.com/tikv/pd/actions/runs/6978834150/job/18990966346
https://github.com/tikv/pd/actions/runs/6973238800/job/18976877533

### Reason for failure (if possible)


### Anything else

--- a/pkg/dashboard/adapter/manager.go
+++ b/pkg/dashboard/adapter/manager.go
@@ -19,6 +19,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pingcap/failpoint"
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb-dashboard/pkg/apiserver"
@@ -75,6 +76,10 @@ func (m *Manager) Stop() {
 func (m *Manager) serviceLoop() {
 	defer logutil.LogPanic()
 	defer m.wg.Done()
+	// TODO: After we fix the atomic problem of config, we can remove this failpoint.
+	failpoint.Inject("skipDashboardLoop", func() {
+		failpoint.Return()
+	})
 
 	ticker := time.NewTicker(CheckInterval)
 	defer ticker.Stop()

--- a/tests/pdctl/config/config_test.go
+++ b/tests/pdctl/config/config_test.go
@@ -24,6 +24,7 @@ import (
 	"time"
 
 	"github.com/coreos/go-semver/semver"
+	"github.com/pingcap/failpoint"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
@@ -60,8 +61,10 @@ func TestConfigTestSuite(t *testing.T) {
 }
 
 func (suite *configTestSuite) TestConfig() {
+	suite.NoError(failpoint.Enable("github.com/tikv/pd/pkg/dashboard/adapter/skipDashboardLoop", `return(true)`))
 	env := tests.NewSchedulingTestEnvironment(suite.T())
 	env.RunTestInTwoModes(suite.checkConfig)
+	suite.NoError(failpoint.Disable("github.com/tikv/pd/pkg/dashboard/adapter/skipDashboardLoop"))
 }
 
 func (suite *configTestSuite) checkConfig(cluster *tests.TestCluster) {

